# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/muzammilsharf/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

*I am choosing Lane 2 (Refresh / Content Opportunity Scoring)*

My goal is to build a reliable decision support system that prioritizes web pages for content updates. This track aligns perfectly with my focus on AI engineering and production pipelines. It allows me to focus deeply on feature engineering, model evaluation, and explainable AI.

I want to build a clean pipeline and prove that a trained machine learning model can prioritize work much better than basic business rules.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The Actor and Decision:
The actors are educational platform owners and publishers. They want to share accurate knowledge and helpful insights with the world. They have limited time and money. They must decide which content to update today and which content to leave alone. This project gives them a clear priority score. It shows them exactly where their pages stand in the market.

The Action:
The content team checks the ranked review queue. When the system flags an underperforming page, they take direct action. They update outdated facts, rewrite weak explanations, and fix search tags. This action helps restore the page's ranking and protects its educational authority.

The Cost of a Wrong Call:

    If the model flags the wrong page: The platform wastes valuable time and money. Writers will spend hours updating content that already works or has zero search demand.

    If the model misses a failing page: Important educational pages will slowly lose search rankings to competitors. The platform loses its visibility in the market. Worse, learners might find outdated information on the site. This damages the platform's authenticity and public trust.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv('../../data/raw/content_refresh_anonymized.csv')

# Display the first few rows to confirm it loaded correctly
print(f"Total rows in dataset: {len(df)}")
df.head()

Total rows in dataset: 30000


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [11]:
#pages that rank on the first page of Google but have terrible click-through rates
rank = (df['avg_position'] <= 10) & (df['ctr'] <= 0.01)
rank.sum()

np.int64(5853)

*Our dataset contains 5,853 pages that already rank on the first page of Google (avg_position <= 10) but suffer from a terrible click rate below 1% (ctr <= 0.01).*

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I Can Claim:

    I can claim this system identifies pages with a high probability of traffic decay.

    I can claim the model ranks review queues more effectively than a static baseline rule.

    I can claim the system provides evidence-based reason codes to guide human reviewers.

What I Cannot Claim:

    I cannot guarantee that updating a recommended page will automatically restore its traffic.

    I cannot claim this model discovers Google algorithm ranking factors.

    I cannot claim this system replaces human editorial judgment. This is a prioritization tool to assist humans, not an automated SEO engine.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.